File: qc-core/src/main/java/com/eager/questioncloud/core/domain/point/service/ChargePointPaymentHistoryService.java
package com.eager.questioncloud.core.domain.point.service;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.point.implement.ChargePointPaymentReader;
import com.eager.questioncloud.core.domain.point.model.ChargePointPayment;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class ChargePointPaymentHistoryService {
    private final ChargePointPaymentReader chargePointPaymentReader;

    public List<ChargePointPayment> getChargePointPayments(Long userId, PagingInformation pagingInformation) {
        return chargePointPaymentReader.getChargePointPayments(userId, pagingInformation);
    }

    public int countChargePointPayment(Long userId) {
        return chargePointPaymentReader.countChargePointPayment(userId);
    }
}

File: qc-core/qc-point/qc-point-api/src/main/kotlin/com/eager/questioncloud/point/service/ChargePointPaymentHistoryService.kt
package com.eager.questioncloud.point.service

import com.eager.questioncloud.common.pagination.PagingInformation
import com.eager.questioncloud.point.dto.ChargePointPaymentHistory
import com.eager.questioncloud.point.implement.ChargePointPaymentReader
import org.springframework.stereotype.Service

@Service
class ChargePointPaymentHistoryService(
    private val chargePointPaymentReader: ChargePointPaymentReader
) {
    fun getChargePointPaymentHistory(userId: Long, pagingInformation: PagingInformation): List<ChargePointPaymentHistory> {
        val chargePointPayments = chargePointPaymentReader.getChargePointPayment(userId, pagingInformation)
        return ChargePointPaymentHistory.from(chargePointPayments)
    }
    
    fun countChargePointPayment(userId: Long): Int {
        return chargePointPaymentReader.countChargePointPayment(userId)
    }
}
